src/Module/Api/Friendica/Events/Create.php

```diff
@@ -0,0 +1,119 @@
+ *
+ */
+
+namespace Friendica\Module\Api\Friendica\Events;
+
+use Friendica\Core\Protocol;
+use Friendica\Core\Worker;
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Model\Event;
+use Friendica\Model\Conversation;
+use Friendica\Model\Item;
+use Friendica\Module\BaseApi;
+use Friendica\Network\HTTPException;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Worker\Delivery;
+
+/**
+ * API endpoint: /api/friendica/event_create
+ */
+class Create extends BaseApi
+{
+	protected function post(array $request = [])
+	{
+		BaseApi::checkAllowedScope(BaseApi::SCOPE_WRITE);
+		$uid = BaseApi::getCurrentUserID();
+
+		// params
+		$request = $this->getRequest([
+			'id'        => 0, //if provided, event will be amended
+			'name'      => '', //summary of the event
+			'desc'      => '', //description in BBCode
+			'startTime' => '', //starttime, required
+			'endTime'   => '', //endtime, required if nofinish false
+			'place'     => '', //location of the event
+			'publish'   => 0,  //publish message
+			'allow_cid' => '', //array of allowed person, if access restricted
+			'allow_gid' => '', //array of allowed groups, if access restricted
+			'deny_cid'  => '', //array of denied person, if access restricted
+			'deny_gid'  => '', //array of denied groups, if access restricted
+		], $request);
+
+		// error if no name specified
+		if (empty($request['name'])) {
+			throw new HTTPException\BadRequestException('event name not specified');
+		}
+
+		// error startDate is not specified
+		if (empty($request['startTime'])) {
+			throw new HTTPException\BadRequestException('startDate not specified');
+		}
+
+		// nofinish if endTime is not specified
+		if (empty($request['endTime'])) {
+			$finish = DBA::NULL_DATETIME;
+			$nofinish = true;
+		} else {
+			$finish = DateTimeFormat::convert($request['endTime'], 'UTC', DI::app()->getTimeZone());
+			$nofinish = false;
+		}
+
+		$start = DateTimeFormat::convert($request['startTime'], 'UTC', DI::app()->getTimeZone());
+
+		// create event
+		$event = [];
+		$event['id']       = $request['id'];
+		$event['uid']      = $uid;
+		$event['summary']  = $request['name'];
+		$event['desc']     = $request['desc'];
+		$event['location'] = $request['place'];
+		$event['start']    = $start;
+		$event['finish']   = $finish;
+		$event['nofinish'] = $nofinish;
+
+		$event['allow_cid'] = $request['allow_cid'];
+		$event['allow_gid'] = $request['allow_gid'];
+		$event['deny_cid']  = $request['deny_cid'];
+		$event['deny_gid']  = $request['deny_gid'];
+		$event['publish']   = $request['publish'];
+
+		$event_id = Event::store($event);
+
+		if (!empty($request['publish'])) {
+			$item = ['network' => Protocol::DFRN, 'protocol' => Conversation::PARCEL_DIRECT, 'direction' => Conversation::PUSH];	
+			$item = Event::getItemArrayForId($event_id, $item);
+			if (Item::insert($item)) {
+				$uri_id = $item['uri-id'];
+			} else {
+				$uri_id = 0;
+			}
+
+			if (!$cid && $uri_id) {
```

Comment:
`$cid` is never declared, which makes the `!$cid` always true and so superfluous.

Reply:
This is why I always declare my variables and initialize them with proper values to avoid problems in the future.

Reply:
This code was copied from another module, which explains the oddity in this case.